pinot-controller/src/main/java/com/linkedin/pinot/controller/validation/ValidationManager.java

```diff
@@ -180,6 +192,22 @@ public void runValidation() {
     LOGGER.info("Validation completed");
   }
 
+  // If we add a new broker, we want to rebuild the broker resource.
+  void rebuildBrokerResourceWhenBrokerAdded(AbstractTableConfig tableConfig, IdealState brokerIdealState) {
```

Comment:
Sounds good! I merged the pull request before receiving your comment, so I've refactored it here: Refactor rebuilding helix entry #983